### PR TITLE
chore(build): support tagged component builds

### PR DESCRIPTION
**Pull request by castastrophe**
## Description

This PR focuses on updating the dependencies for all component packages and migrates the build scripts into a separate project.json for each component. The build is still leveraging gulp but it's firing it from the root of the package, thus removing the need for a local component dependency on gulp and making it easier to update the build in a central location.

The mod-extractor was updated to run on an individual package rather than the entire components folder so that it could be chained with the build task.

## How and where has this been tested?

No regressions for the following commands @mdt2:

- [x] `yarn dev`
- [x] `yarn start`
- [x] `yarn build:preview`
- [x] `yarn build`
  - my understanding is that this replaces `yarn build:components` - mdt2
- [x] `yarn clean`
  - my understanding is that this replaces `yarn clean:components` - mdt2

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] If my change impacts **other components**, I have tested to make sure they don't break.
- [x] If my change impacts **documentation**, I have updated the documentation accordingly.
- [x] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: package.json
Comment (castastrophe) [inline package.json:56]:
```suggestion
    "prestart": "yarn build",
    "start": "NODE_ENV=development yarn workspace @spectrum-css/preview start",
```
```suggestion
    "start": "NODE_ENV=development nx start @spectrum-css/preview",
```

**Comment by castastrophe** [comment]
Debugging the vrt regressions in the most recent build.

File: components/commons/package.json
Comment (mdt2) [inline components/commons/package.json:16]: Can you tell me about this change?

File: components/logicbutton/metadata/mods.md
Comment (mdt2) [inline components/logicbutton/metadata/mods.md:3]: My assumption is that we're seeing changes to the mods file for some of these because these may have been accidentally excluded before, is that right? Should these be included?

File: components/menu/package.json
Comment (mdt2) [inline components/menu/package.json:37]: What makes these optional and `@spectrum-css/icon` not optional?

**Review by mdt2 — COMMENTED** [review]
The commands are running well. Mostly just have some context questions for now 🙂

**Review by pfulton — APPROVED** [review]
I was able to pull this down and run through the verification steps. Big improvement on being able to run the different commands without first having to run `yarn cache:clean`!

File: components/actionbutton/metadata/mods.md
Comment (castastrophe) [reply components/actionbutton/metadata/mods.md:54]: Yeah, I didn't want to mess too much with the script because I knew Josh's was going to merge soon but I needed to update it enough to run on individual packages so it could be a part of the build flow.

File: components/actionbutton/package.json
Comment (castastrophe) [reply components/actionbutton/package.json:25]: Since none of our packages include `dependencies`, none of these changes will impact them downstream. All our packages list `devDependencies` (those required for the build process) and `peerDependencies` (expected to be installed by the downstream consumer). We're clarifying the existing `peerDependencies` in this PR to highlight elements that are optional to rendering the component so that downstream users not using those variations don't get unnecessary warnings for missing peer dependencies.

File: components/actiongroup/package.json
Comment (castastrophe) [reply components/actiongroup/package.json:28]: None of our packages need `devDependencies` anymore because the build work happens at root now. The only reason we had them previously was for storybook but storybook doesn't need them either in a monorepo environment.

File: components/commons/package.json
Comment (castastrophe) [reply components/commons/package.json:16]: Yes, so commons doesn't actually have a compiled output because it's a pre-compilation utility with extends in it. When someone tries to access `@spectrum-css/commons` it was sending them to a non-existent package and so would always throw an error. Now if you want to use our extends in another project, you can `require("@spectrum-css/commons")` and load up all the extends we define here.

File: components/logicbutton/metadata/mods.md
Comment (castastrophe) [reply components/logicbutton/metadata/mods.md:3]: These are similar to what Josh's update was seeing because we weren't running on compiled results.

File: components/menu/package.json
Comment (castastrophe) [reply components/menu/package.json:37]: I think in the markup I saw it included Icon and it wasn't wrapped in an if or when so I left it off optional.  I'd defer after merge to the developer who worked on menu last to make that final call.

File: components/actiongroup/package.json
Comment (mdt2) [reply components/actiongroup/package.json:28]: Oh ok, and since this was never a peer dependency it doesn't need to be one now 👍

File: components/commons/package.json
Comment (mdt2) [reply components/commons/package.json:16]: Cool cool, thanks for explaining 🙂

**Review by jawinn — COMMENTED** [review]
This LGTM. I was able to run all the commands, and tested making changes to CSS and docs while dev and start were running. I did notice one warning in the console output after `yarn build:preview`: 

`(node:46827) [DEP_WEBPACK_STATS_WARNINGS_FILTER] DeprecationWarning: config.stats.warningsFilter is deprecated in favor of config.ignoreWarnings`

**Pull request merged**
